crates/signer-ledger/src/signer.rs

```diff
@@ -43,7 +46,33 @@ impl alloy_network::TxSigner<Signature> for LedgerSigner {
         &self,
         tx: &mut dyn SignableTransaction<Signature>,
     ) -> Result<Signature> {
-        sign_transaction_with_chain_id!(self, tx, self.sign_tx_rlp(&tx.encoded_for_signing()).await)
+        let encoded = tx.encoded_for_signing();
+
+        match encoded.as_slice() {
+            // Ledger requires passing EIP712 data to a separate instruction
+            #[cfg(feature = "eip712")]
+            [0x19, 0x1, data @ ..] => {
+                let domain_sep = data
+                    .get(..32)
+                    .ok_or_else(|| {
+                        Error::other("eip712 encoded data did not have a domain separator")
```

Comment:
could you use the full path here so we don't need to feature gate the error import